docs/API.md

```diff
@@ -0,0 +1,176 @@
+# Shipyard API
+
+Logically, the API has three parts to handle the three areas of functionality in Shipyard.
+
+1. Document Staging
+2. Action Handling 
+3. Airflow Monitoring 
+
+There are several standard error responses that should be handled appropriately by a client of this API:
+* 401 Authentication Error  
+If not authenticated
+* 403 Forbidden  
+If not authorized
+* 404 Not found  
+If a variable in the path doesn't reference a valid resource 
+* 503 Service Unavailable  
+Error if an upstream system cannot be reached (e.g. Deckhand for documents, Airflow for actions)
+
+---
+## <a name="DocumentStagingAPI"></a> Document Staging API
+Shipyard will serve as the entrypoint for documents and secrets into a site. At any point in time, there will be two represented versions of documents in a site that are accessible via this API. 
+
+* The lastDeployed version, which represents the last version of documents that successfully completed deployment.
+* The staged version, which represents the document set that has been ingested by this API since the lastDeployed version.
+
+### /v1.0/configDocs  
+Represents the site configuration documents.  
+
+#### POST 
+Ingests site configuration documents. Synchronous.
+##### Responses
+* 200 OK  
+If the documents are successfully ingested, even with validation failures. Response message includes:
+  * a link to the GET /configDocs 
+  * a list of validation results
+
+#### GET 
+Returns the source documents for the most recently staged version
+Query Parameter "lastDeployed=true" will insetead return the documents for the currently deployed version.
+##### Responses
+* 200 OK  
+If documents can be retrieved.
+
+#### DELETE  
+Updates the configDocs to be the lastDeployed versions in deckhand, effectively discarding any staged documents.
+###### Responses
+* 204 No Content
+
+
+### /v1.0/secrets
+Represents the secrets documents that will be used in combination with the config docs.  
+
+##### POST 
+Ingest new secrets documents.
+###### Responses
```

Comment:
We should add samples for the Payloads here, or include a link to the format in the Deckhand document.

There is a 2 Stage notion when it comes to secrets. 
1) Create the Secret/Certificate. Update is a form of creation as well. But the idea is that this does not mean the secrecy is configured or in use yet. It simple means its now available in Deckhand.
2) Apply the Secret to its intended target. We talked about most of this been basically updates to chart that use the secrets or certificates. But we did discuss some "types of Certs or Secrets" that maybe do not follow the same path. Can't remember if we agreed that all of those would be taken care of with some "Dummy" Charts. If tat is true then this all spiel can be ignored. Otherwise , we would need a mechanism(api)  to say "Apply this Cert or Secret".